src/pages/gd/Swap/SwapConfirmModal/index.tsx

```diff
@@ -65,28 +65,67 @@ function SwapConfirmModal({
     const web3 = useWeb3()
     const [status, setStatus] = useState<'PREVIEW' | 'CONFIRM' | 'SENT'>('SENT')
     const [hash, setHash] = useState('')
+    const [sender, setSender] = useState('')
+
+    const senderRef = useRef(sender)
+
+    useEffect(() => {
+      senderRef.current = sender
+    }, [sender])
 
     const handleSwap = async () => {
         setStatus('CONFIRM')
+
+        const prepareTx = (from: string) => {
+          setSender(from)
+        }
+        
         const onSent = (hash: string) => {
             setStatus('SENT')
             setHash(hash)
-            if (onConfirm) onConfirm()
-        }
-        try {
-            let transactionDetails = buying ? await buy(web3!, meta!, onSent) : await sell(web3!, meta!, onSent)
+            const inputSig = meta?.inputAmount.toSignificant(5)
+            const minimumOutputSig = meta?.minimumOutputAmount.toSignificant(5)
+
+            const tradeInfo = {
+              input: {
+                decimals: meta?.inputAmount.currency.decimals,
+                symbol: meta?.inputAmount.currency.symbol
+              },
+              output: {
+                decimals: meta?.outputAmount.currency.decimals,
+                symbol: meta?.outputAmount.currency.symbol
+              }
+            }
+            const summary = 'Swapping ' + inputSig + ' ' 
+                            + meta?.inputAmount.currency.symbol + 
+                            ' to a minimum of ' + minimumOutputSig + ' ' +
+                            meta?.outputAmount.currency.symbol
             globalDispatch(
-                addTransaction({
-                    chainId: chainId!,
-                    hash: transactionDetails.transactionHash,
-                    from: transactionDetails.from
-                })
+              addTransaction({
+                chainId: chainId!,
+                hash: hash,
+                from: senderRef.current,
```

Comment:
can you explain why a ref is required? this logic means that if a new sender is set all TXs would now point to the new sender, doesnt make much sense

Reply:
Because setSender(from) doesn't take effect until after the transaction has been submitted/function finished
so what you then get every TX is that the previous sender set is used as from.
What you get when it is your first transaction, is an empty sender

This is the only solution that I know of for now who takes the currently set sender (within this flow)
(setSender from prepareTx)

But, with that being said.

would have been easier if I from the get go did this:
`req.on(transactionHash, (hash: string) => onSent(hash, account)`
 
 so the prepareTx isn't necessary at all..